src/services/branches/useBranchBundlesNames.tsx

```diff
@@ -0,0 +1,177 @@
+        .nullable(),
+    })
+    .nullable(),
+})
+
+const query = `query BranchBundlesNames(
+  $owner: String!
+  $repo: String!
+  $branch: String!
+) {
+  owner(username: $owner) {
+    repository(name: $repo) {
+      __typename
+      ... on Repository {
+        branch(name: $branch) {
+          head {
+            bundleAnalysisReport {
+              __typename
+              ... on BundleAnalysisReport {
+                bundles {
+                  name
+                }
+              }
+              ... on MissingHeadReport {
+                message
+              }
+            }
+          }
+        }
+      }
+      ... on NotFoundError {
+        message
+      }
+      ... on OwnerNotActivatedError {
+        message
+      }
+    }
+  }
+}`
+
+interface UseBranchBundlesNamesArgs {
+  provider: string
+  owner: string
+  repo: string
+  branch?: string
+  opts?: {
+    enabled?: boolean
+  }
+}
+
+export const useBranchBundlesNames = ({
+  provider,
+  owner,
+  repo,
+  branch: branchArg,
+  opts = {},
+}: UseBranchBundlesNamesArgs) => {
+  const { data: repoOverview, isSuccess } = useRepoOverview({
+    provider,
+    repo,
+    owner,
+  })
+
+  const branch = branchArg ?? repoOverview?.defaultBranch
+
+  return useQuery({
+    queryKey: ['BranchBundlesNames', provider, owner, repo, branch],
+    queryFn: ({ signal }) =>
+      Api.graphql({
+        provider,
+        query,
+        signal,
+        variables: {
+          owner,
+          repo,
+          branch,
+        },
+      }).then((res) => {
+        const parsedData = BranchBundleSummaryDataSchema.safeParse(res?.data)
+
+        if (!parsedData.success) {
+          return Promise.reject({
+            status: 404,
+            data: {},
+          })
+        }
+
+        const data = parsedData.data
+
+        if (data?.owner?.repository?.__typename === 'NotFoundError') {
+          return Promise.reject({
+            status: 404,
+            data: {},
+          })
+        }
+
+        if (data?.owner?.repository?.__typename === 'OwnerNotActivatedError') {
+          return Promise.reject({
+            status: 403,
+            data: {
+              detail: (
+                <p>
+                  Activation is required to view this repo, please{' '}
+                  {/* @ts-expect-error */}
+                  <A to={{ pageName: 'membersTab' }}>click here </A> to activate
+                  your account.
+                </p>
+              ),
+            },
+          })
+        }
+
+        let bundles: Array<string> = []
+        if (
+          data?.owner?.repository?.branch?.head?.bundleAnalysisReport
+            ?.__typename === 'BundleAnalysisReport'
+        ) {
+          bundles =
+            data.owner.repository.branch.head.bundleAnalysisReport.bundles?.map(
+              ({ name }) => name
+            )
+        }
+
+        return { bundles }
+      }),
+    enabled: (isSuccess && isString(branch)) || opts?.enabled,
```

Comment:
Can you explain the first condition please? this looks to me like it will be resolved AFTER the hook was resolved

Reply:
So this is to disable the query if no branch was passed, because we first need to make a call to grab the default branch from repo overview, which is why we're checking to see if the request was successful from repo overview and we have a valid branch, if we don't then we shouldn't run this query.

Reply:
ohhhh isSuccess is coming from repoOverview, makes sense